week-1-ts-ecosystem/w01e02-enter-the-world-of-typescript/src/framework/dom-creators.ts

```diff
@@ -0,0 +1,21 @@
+function createElement<Element extends HTMLElement>(
+  tagName = 'div',
+  className = ''
+): Element {
+  const element = document.createElement(tagName);
+
+  if (className.length) {
+    element.className = className;
+  }
+
+  return element as Element;
```

Comment:
To można otypować lepiej bez asercji typów.

Wystarczy że Twoim generykiem nie będzie HTMLElement, tylko właśnie `tagName`.
Aktualnie to `string` jako `"div"` i ok, ale może to być jeden z obsługiwanych "stringów / typów" jako unia:

```typescript
function createElement<TAG extends 'div' | 'p' | 'section' | 'article'>(tagName: TAG, ....

```

I to powinno sie dobrze zachowywać bez potrzeby `as`

Reply:
Sprawdź proszę ten zapis powyżej u siebie w kodzie. powinien zadziałać z pudełka.